tests/rptest/scale_tests/extreme_recovery_test.py

```diff
@@ -0,0 +1,237 @@
+    """
+    This test case covers normal recovery process at a larger scale, using a
+    bigger cluster and creating more partitions.
+    """
+    # TODO until we have weekly/bi-weekly schedules, this test
+    # is scaled down to run with CDT nightlies.
+    NIGHTLY_SCALE_DOWN = 2
+
+    # TODO rookie numbers
+    NODE_COUNT = 5
+    # More partitions cause the test take a long time to run, hitting some
+    # bottlenecks in the existing test, such as serial computation of file
+    # checksums, etc.
+    PARTITONS = NODE_COUNT * 200 // NIGHTLY_SCALE_DOWN
+    NUM_TOPICS = 2
+    PRODUCERS_PER_TOPIC = 1
+    MESSAGE_COUNT = 50 * 1000 * PARTITONS // NIGHTLY_SCALE_DOWN
+    MSG_SIZE = 1023
+
+    # Scale tests run ducktape on dedicated nodes, so we can expect some
+    # minimum bandwidth, and set a test timeout based on that.
+    _min_bandwidth_bps = 2**26  # 64 MiB/sec
+    _total_bytes = MSG_SIZE * MESSAGE_COUNT
+    expected_transfer_sec = _total_bytes // (_min_bandwidth_bps) + 30
+    _msg_per_topic = MESSAGE_COUNT // NUM_TOPICS
+    part_per_topic = PARTITONS // NUM_TOPICS
+
+    _status_log: list[str] = []
+
+    def __init__(self, test_context: TestContext, s3_client: S3Client,
+                 kafka_tools: KafkaCliTools, rpk_client: RpkTool,
+                 s3_bucket: str, logger: Logger, topics: Sequence[TopicSpec],
+                 rpk_producer_maker: Callable):
+        self.topics = topics
+        self.ctx = test_context
+        self.logger = logger
+        self.redpanda = kafka_tools._redpanda
+        super(RecoveryScale,
+              self).__init__(s3_client, kafka_tools, rpk_client, s3_bucket,
+                             logger, rpk_producer_maker)
+
+    @property
+    def verify_s3_content_after_produce(self):
+        # Data verification is good here, can catch missing uploads.
+        return True
+
+    def _status(self, message: str) -> None:
+        self.logger.info(message)
+        self._status_log.append(message)
+
+    def log_status(self):
+        self.logger.info("=====  \u2696  Scale Test Summary  \u2696  =====")
+        for l in self._status_log:
+            self.logger.info(l)
+        self.logger.info("=====  \u2696                      \u2696  =====")
+
+    def generate_baseline(self):
+        producers: list[FranzGoVerifiableProducer] = []
+        self._status(
+            f"{len(self.topics)} topics, {RecoveryScale.PARTITONS} partitions")
+        t0 = time()
+        # start multiple producers per topic
+        for topic in self.topics:
+            for i in range(self.PRODUCERS_PER_TOPIC):
+                self.logger.info(
+                    f"gen baseline: make producer {i+1} of "
+                    f"{self.PRODUCERS_PER_TOPIC} for {topic.name}")
+                p = FranzGoVerifiableProducer(self.ctx, self.redpanda,
+                                              topic.name, self.MSG_SIZE,
+                                              self._msg_per_topic)
```

Comment:
question: maybe I misunderstood the calculations, but should this `self._msg_per_topic` be divided by `self.PRODUCERS_PER_TOPIC` when building the producer, so it only produces its share of messages?

Reply:
Good point. Although the divisor is currently `1`, this should be updated in case someone changes the constants in the future. I'll add a divide now.

Reply:
Fixed in latest update.